homeassistant/components/vacuum/xiaomi_miio.py

```diff
@@ -383,3 +399,24 @@ def update(self):
             _LOGGER.error("Got OSError while fetching the state: %s", exc)
         except DeviceException as exc:
             _LOGGER.warning("Got exception while fetching the state: %s", exc)
+
+    async def async_clean_zone(self,
+                               zone: list,
+                               repeats: int = 1):
+        """Clean selected area for the number of repeats indicated."""
+        repeats = max(1, min(3, repeats))
+        _LOGGER.debug("Zone to clean: %s repeats: %s", zone, repeats)
+        from miio import DeviceException
+        _LOGGER.debug("Original zone: %s", zone)
+        for _zone in zone:
+            _zone.append(repeats)
+        _LOGGER.debug("Zone with repeats: %s", zone)
+        try:
+            await self.hass.async_add_executor_job(
+                self._vacuum.zoned_clean(zone))
```

Comment:
Separate the function from the argument with a comma, and remove the call parenthesis, ie don't call the method here. That will be done by `hass.async_add_executor_job`.

Reply:
fixed